livekit-agents/livekit/agents/job.py

```diff
@@ -291,6 +292,106 @@ async def connect(
 
         _apply_auto_subscribe_opts(self._room, auto_subscribe)
 
+    def _create_task_with_future(
+        self, task_id: str, coro: Coroutine[Any, Any, Any]
+    ) -> asyncio.Future[Any]:
+        """Create a task and return a future that resolves when the task completes."""
+        if task_id in self._pending_tasks:
+            logger.warning(f"Task {task_id} already exists, returning existing future")
+            return self._pending_tasks[task_id]
+
+        fut = asyncio.Future[Any]()
+        task = asyncio.create_task(coro, name=task_id)
+
+        def _on_task_done(t: asyncio.Task[Any]):
+            self._pending_tasks.pop(task_id)
+            if t.exception():
+                fut.set_exception(t.exception())
+            else:
+                fut.set_result(t.result())
+
+        task.add_done_callback(_on_task_done)
+        self._pending_tasks[task_id] = fut
+        return fut
+
+    def disconnect(self) -> asyncio.Future[None]:
+        """Disconnects the agent from the room, but does not delete the room."""
+        return self._create_task_with_future("disconnect", self._room.disconnect())
+
+    def delete_room(self) -> asyncio.Future[None]:
+        """Deletes the room and disconnects all participants."""
+        return self._create_task_with_future(
+            "delete_room", self.api.room.delete_room(api.DeleteRoomRequest(room=self._room.name))
+        )
+
+    def get_sip_participants(self) -> list[rtc.RemoteParticipant]:
+        return [
+            p
+            for p in self._room.remote_participants.values()
+            if p.kind == rtc.ParticipantKind.PARTICIPANT_KIND_SIP
+        ]
+
+    def add_sip_participant(
+        self,
+        sip_call_to: str,
+        sip_trunk_id: str,
+        participant_identity: str,
+        participant_name: str,
+    ) -> asyncio.Future[api.SIPParticipantInfo]:
+        """
+        Add a SIP participant to the room.
+
+        Make sure you have an outbound SIP trunk created in LiveKit.
+        See https://docs.livekit.io/sip/trunk-outbound/ for more information.
+        """
+        return self._create_task_with_future(
+            f"add_sip_participant-{sip_call_to}",
+            self.api.sip.create_sip_participant(
+                api.CreateSIPParticipantRequest(
+                    room_name=self._room.name,
+                    participant_identity=participant_identity,
+                    sip_trunk_id=sip_trunk_id,
+                    sip_call_to=sip_call_to,
+                    participant_name=participant_name,
+                )
+            ),
+        )
+
+    def transfer_sip_participant(
+        self,
+        participant: rtc.RemoteParticipant,
```

Comment:
we can accept either identity or object

```suggestion
        participant: rtc.RemoteParticipant | str,
```